test/tests/functional/pbs_cray_smoketest.py

```diff
@@ -0,0 +1,186 @@
+        vname = j1.get_vnodes()[0]
+        self.server.expect(NODE, {ATTR_rescavail + '.vntype': 'cray_login'},
+                           id=vname)
+
+        cu = CrayUtils()
+        # Check if number of compute nodes in use are 0
+        self.assertEqual(cu.count_node_in_state('use'), 0)
+
+        # verify the contents of output/error files
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid1, extend='x')
+        error_file = os.path.join(sub_dir, 'cray_login.e' + jid1.split('.')[0])
+        self.assertEqual(os.stat(error_file).st_size, 0,
+                         msg="Job error file should be empty")
+
+        output_file = os.path.join(
+            sub_dir, 'cray_login.o' + jid1.split('.')[0])
+        with open(output_file, 'r') as of:
+            for line in of:
+                self.assertTrue("Hello World" in line,
+                                msg="Job output file incorrect")
+                break
+
+    @tags('cray', 'smoke')
+    def test_cray_compute_job(self):
+        """
+        Submit a simple sleep job that runs on a compute node and
+        expect the job to go in running state on a compute node.
+        Verify that the job runs to completion and check job output/error.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'})
+        j1 = Job(TEST_USER, {ATTR_l + '.vntype': 'cray_compute',
+                             ATTR_N: 'cray_compute'})
+
+        scr = []
+        scr += ['echo Hello World\n']
+        scr += ['sleep 5\n']
+        scr += ['aprun -B sleep 10\n']
+
+        sub_dir = self.du.mkdtemp(uid=TEST_USER.uid)
+        j1.create_script(scr)
+        jid1 = self.server.submit(j1, submit_dir=sub_dir)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        # fetch node name where the job is running and check that the
+        # node is a compute node
+        self.server.status(JOB, 'exec_vnode', id=jid1)
+        vname = j1.get_vnodes()[0]
+        self.server.expect(NODE, {ATTR_rescavail + '.vntype': 'cray_compute'},
+                           id=vname)
+        # Sleep for some time before aprun actually starts
+        # using the reservation
+        time.sleep(6)
+
+        cu = CrayUtils()
+        # Check if number of compute nodes in use is 1
+        self.assertEqual(cu.count_node_in_state('resv'), 1)
+        if self.du.get_platform() == 'cray':
+            # Cray simulator will not show anything in 'use' because
+            # aprun command is just a pass through on simulator
+            self.assertEqual(cu.count_node_in_state('use'), 1)
+        # verify the contents of output/error files
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid1, extend='x')
+        error_file = os.path.join(
+            sub_dir, 'cray_compute.e' + jid1.split('.')[0])
+        self.assertEqual(os.stat(error_file).st_size, 0,
+                         msg="Job error file should be empty")
```

Comment:
Have you run this on a Cray ALPS simulator?  I would expect there would be an error due to the aprun command not being found.

Reply:
I made changes and ran them in the Cray simulator. I skip this test on the Cray simulator due to the aprun not being found.